### ISSUE TITLE
Improve affiliate deletion screen

### ISSUE THREAD

**Issue by amdrew** [AffiliateWP/AffiliateWP#2546]
Checking the `Delete this affiliate’s user account as well?` in the screenshot below will delete any post associated with the user account (as per https://codex.wordpress.org/Function_Reference/wp_delete_user).

At first glance, it may also appear that the checkbox is to confirm the deletion of the affiliate, not anything related to the user account (since they are so close together). This could also be attributed to the `Deleting this affiliate will...` text above the checkbox, and then the first few words of the checkbox option, `Delete this affiliate...`. 

<img width="454" alt="screenshot 2018-01-31 13 05 35" src="https://user-images.githubusercontent.com/52581/35598959-492bebba-068b-11e8-8541-2dd57982f468.png">

I propose we change the wording of the checkbox to be clearer on what enabling it will do, in addition to just the user account.

We could expand on this further and highlight the checkbox section in some way that better illustrates the seriousness of that specific option.

A customer accidentally deleted an admin account and all pages on their site attributed to that admin were deleted.

**Comment by pippinsplugins** [comment]
I'm not sure we should ever delete associated content. I cannot think of a scenario where an admin would want that to happen.

**Comment by amdrew** [comment]
Here's what I propose:

On our screen, if `Delete this affiliate’s user account as well?` is enabled and `Confirm Deletion` is clicked, we redirect to the standard WordPress delete user page:

<img width="301" alt="screenshot 2018-02-02 12 19 57" src="https://user-images.githubusercontent.com/52581/35708556-8e6e7e16-0813-11e8-8517-c39f7bb4a49d.png">

This gives the site admin _one_ last chance to confirm the deletion of that user account. It's not our job to build out a custom UI within AffiliateWP to handle this, so lets allow WordPress to take over. 

We should inject a AfifliateWP related notice to the top of the page, letting the admin know the affiliate was deleted but they have one more step to confirm.

The above shows what happens when the user does not have any posts associated to them. Here's what the same page shows when they do:

<img width="408" alt="screenshot 2018-02-02 12 19 28" src="https://user-images.githubusercontent.com/52581/35708673-15035fdc-0814-11e8-9705-809d5fd6a5aa.png">

**Comment by amdrew** [comment]
Multisite shows a slightly different screen with added messages at the top and bottom and site the user is assigned to:

<img width="622" alt="screenshot 2018-02-02 12 26 51" src="https://user-images.githubusercontent.com/52581/35708748-6c5d9f90-0814-11e8-832c-eb0c5175935f.png">

**Comment by pippinsplugins** [comment]
I like that proposal.

**Comment by pippinsplugins** [comment]
@tubiz would you like to work on this?

**Comment by tubiz** [comment]
@pippinsplugins yes.

**Comment by tubiz** [comment]
@pippinsplugins @amdrew on a multisite install user account(s) can only be removed from the site in question and not deleted

<img width="500" alt="multisite remove user from site" src="https://user-images.githubusercontent.com/602184/36485665-f4d055e0-171c-11e8-8971-7ff602173416.png">

Deleting the user account(s) can only be done in the users page on the network admin by a user with the super admin privileges

<img width="672" alt="multisite delete user from site" src="https://user-images.githubusercontent.com/602184/36485795-47708d6a-171d-11e8-8396-d383ba73113f.png">
 
When the `Delete this affiliate’s user account as well?` checkbox is checked on a multisite install what should be done
- User account(s) removed from the current site?
- User account(s) deleted from all networks and sites?

**Comment by pippinsplugins** [comment]
I don't think we should show `Delete this affiliate’s user account as well?` when on multisite. We can include a note like `To also delete this account, go to the Users page of the Network Admin.`

**Comment by tubiz** [comment]
PR: #2585 

<img width="1270" alt="affiliate delete notice" src="https://user-images.githubusercontent.com/602184/36519168-5156d95c-178a-11e8-825b-8a18971f0952.png">

<img width="641" alt="multisite delete affiliate" src="https://user-images.githubusercontent.com/602184/36519167-512b08e0-178a-11e8-913a-0bcf7d1fc627.png">

**Comment by DrewAPicture** [comment]
Hmm, I'm just not a huge fan of dipping into core territory here. I kind of get the whole ease-of-use argument being made, but we're effectively reinventing the user-deletion wheel and stepping outside of the AffiliateWP realm to accomplish it here. It would be fair to say that we're already overstepping by creating users in some contexts, but it still feels a little icky.

I feel like we'd have to recreate a lot of the logic already handled in core for the more complex cases, such as deletion in multisite.

This is mostly because we'll have to put in all of the checks for super admin and whether the user can be deleted across the network and what happens if their content on one or more many sites has to be reattributed to a user that may not be a member of all those sites, etc. I think it can quickly get much more complicated and a pain to maintain what core is already set up to do.
